ProyectoSemestral3/Assets/Scripts/PlayerMovement.cs

```diff
@@ -4,40 +4,24 @@
 
 public class PlayerMovement : MonoBehaviour
 {
-    public CharacterController controller;
-    private Vector3 playerVelocity;
-    private bool isitGrounded;
-    public float speed = 5;
-    public float jumpHeight = 1f;
-    public float gravity = -9.81f;
+    [SerializeField] private CharacterController _controller;
+    [SerializeField] private Transform cam;
+    [SerializeField] private float _speed = 6f;
+    [SerializeField] private float _turnTime = 0.1f;
+    float _turnVelocity;
 
-    private void Start()
+    private void Update()
     {
-        controller = gameObject.AddComponent<CharacterController>();
-    }
-    
-    void Update()
-    {
-        isitGrounded = controller.isGrounded;
-        if (isitGrounded && playerVelocity.y < 0)
-        {
-            playerVelocity.y = 0f;
-        }
-
-        Vector3 move = new Vector3(Input.GetAxis("Horizontal"), 0, Input.GetAxis("Vertical"));
-        controller.Move(move * Time.deltaTime * speed);
-
-        if (move != Vector3.zero)
+        float horizontal = Input.GetAxisRaw("Horizontal");
+        float vertical = Input.GetAxisRaw("Vertical");
+        Vector3 direction = new Vector3(horizontal, 0f, vertical).normalized;
+        if(direction.magnitude >= 0.1f)
```

Comment:
Change `if(` to `if (`, with space between.
Also move the `0.1f` to be a constant or a field on the inspector, this is a magic number.